academy/xl-run-prod/4-network.md

```diff
@@ -7,16 +7,171 @@ tag: deep-dive
+
+```txt
+ce1c54ea7a2c50b4b9f2f869faf8fa4d1a1cf43a
+```
+
+If you lose `node_key.json`, or have it stolen, it is not as serious as if you lost your token's private key. Your node can always recreate it and let your peers know about the new key, no hard feelings. Its location is mentioned in `config.toml` on the line `node_key_file = "config/node_key.json"`.
+
+The node key also exists so that your own node can identify itself if it tried to connect to itself via a circuitous peer-to-peer route, and therefore ought to cut the useless connection.
+
+In short here is the information you need to share with other early participants in the network:
+
+* Listen address, for instance: `"tcp://74.6.231.20:26656"`.
+* Node key, for instance: `ce1c54ea7a2c50b4b9f2f869faf8fa4d1a1cf43a`.
+
+The shorthand for this information is written and exchanged as such:
+
+```txt
+ce1c54ea7a2c50b4b9f2f869faf8fa4d1a1cf43a@74.6.231.20:26656
+```
+
+If you create a node for a network that is already running, you need to do the same above steps but you do not need to inform others of your parameters, because when you connect, your node will do anyway.
+
+As a side-note, your computer or local network may not allow other nodes to initiate a connection to your node on port `26656`. So it is a good idea to open this port in the firewall(s).
+
+## Connection to others
+
+You collected your node's information. You shared it with the early network participants. In return you got theirs. You can put those in `config.toml` separated by commas:
+
+```toml
+persistent_peers = "432d816d0a1648c5bc3f060bd28dea6ff13cb413@216.58.206.174:26656,
+5735836cbaa747e013e47b11839db2c2990b918a@121.37.49.12:26656"
+```
+
+If one of the operators informs you that they node behaves as a seed node, then you add it under:
+
+<CodeGroup>
+
+<CodeGroupItem title="Up to v0.34" active>
+
+```toml
+seeds = "432d816d0a1648c5bc3f060bd28dea6ff13cb413@216.58.206.174:26656"
+```
+
+</CodeGroupItem>
+
+<CodeGroupItem title="From v0.35">
+
+```toml
+bootstrap_peers = "432d816d0a1648c5bc3f060bd28dea6ff13cb413@216.58.206.174:26656"
+```
+
+</CodeGroupItem>
+
+</CodeGroup>
+
+You can also take this opportunity to document the list of peers on your _production_ repository (the same that hosts the genesis file).
+
+Note that you are not obliged to put all the known peers in your `persistent_peers`. You may well choose to put them only those you trust.
+
+## Further network configuration
+
+Setting up your peer and identifying other peers is important. However this is not the only network configuration available. Look into `~/.myprojectd/config/config.toml` for tweaks.
+
+If you change parameters in this file, you are not going to affect the ability of the network to reach consensus on blocks. Parameters that are necessary for consensus are all in the genesis file.
+
+[Parameters in `config.toml`](https://docs.tendermint.com/master/nodes/configuration.html) can be divided into two broad categories:
```

Comment:
Tendermint advice to not put any link to `/master` but pin the version. The `/master` endpoint does not exist anymore.

```suggestion
[Parameters in `config.toml`](https://docs.tendermint.com/v0.34/tendermint-core/configuration.html) can be divided into two broad categories:
```